core/src/main/scala/org/apache/spark/util/AsynchronousListenerBus.scala

```diff
@@ -90,8 +93,9 @@ private[spark] abstract class AsynchronousListenerBus[L <: AnyRef, E](name: Stri
    * listens for any additional events asynchronously while the listener bus is still running.
    * This should only be called once.
    */
-  def start() {
+  def start(sc: SparkContext) {
```

Comment:
Add an `@param sc Used to stop the SparkContext in case the listener thread dies.`, or something to that effect